File: array/0896_monotonic_array/0896_monotonic_array.go
package main

import "fmt"

func isMonotonic(A []int) bool {
	if len(A) <= 2 {
		return true
	}
	noInc, noDec := true, true
	for i := 0; i < len(A)-1; i++ {
		if A[i] > A[i+1] {
			noDec = false
		}
		if A[i] < A[i+1] {
			noInc = false
		}

	}
	return noDec || noInc
}

func main() {
	A := []int{1, 2, 2, 3}
	res := isMonotonic(A)
	fmt.Println(res)
}

File: array/0896_monotonic_array/0896_monotonic_array.py
from typing import List

class Solution:
    def isMonotonic(self, A: List[int]) -> bool:
        cnt_inc = 0; cnt_dec = 0
        for i, v in enumerate(A):
            if i == 0: prev = v; continue
            if   v > prev: cnt_inc += 1
            elif v < prev: cnt_dec += 1
            if cnt_inc and cnt_dec: return False
            prev = v
        return True


A = [1,2,2,3]
res = Solution().isMonotonic(A)
print(res)